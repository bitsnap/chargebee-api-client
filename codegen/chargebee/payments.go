package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListTransactions generates chargebee client code to fetch all transactions
//
// API: https://{site}.chargebee.com/api/v2/transactions
//
// Documentation: https://apidocs.chargebee.com/docs/api/transactions?lang=curl#list_transactions
func GenerateListTransactions() string {
	return client.GenerateList("Transactions", "Transaction", "chargebee.com/api/v2/transactions")
}

// GenerateRetrieveTransaction generates chargebee client code to retrieve specific transaction
//
// API: https://{site}.chargebee.com/api/v2/transactions/{transaction-id}
//
// Documentation: https://apidocs.chargebee.com/docs/api/transactions?lang=curl#retrieve_an_transaction
func GenerateRetrieveTransaction() string {
	return client.GenerateRetrieve("Transaction", "chargebee.com/api/v2/transactions")
}

// GenerateTransactionsModel generates chargebee Transaction domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/transactions?lang=curl#transaction_attributes
func GenerateTransactionsModel() string {
	return client.GenerateModel("Transaction", "Transaction attributes", "https://apidocs.chargebee.com/docs/api/transactions")
}
