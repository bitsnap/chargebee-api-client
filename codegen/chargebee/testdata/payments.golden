# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee payment transactions code generation when generating Chargebee API client should be able to generate payment transactions models" = '''
(string) (len=4444) "type Transaction struct {\n    Id string `json:\"id\" validate:\"required\"`\n    CustomerId string `json:\"customer_id\"`\n    SubscriptionId string `json:\"subscription_id\"`\n    GatewayAccountId string `json:\"gateway_account_id\"`\n    PaymentSourceId string `json:\"payment_source_id\"`\n    PaymentMethod enums.PaymentMethodEnum `json:\"payment_method\" validate:\"required\"`\n    ReferenceNumber string `json:\"reference_number\"`\n    Gateway enums.GatewayEnum `json:\"gateway\" validate:\"required\"`\n    Type enums.TypeEnum `json:\"type\" validate:\"required\"`\n    Date *time.Time `json:\"date\"`\n    SettledAt *time.Time `json:\"settled_at\"`\n    ExchangeRate *big.Float `json:\"exchange_rate\"`\n    CurrencyCode string `json:\"currency_code\" validate:\"required\"`\n    Amount uint64 `json:\"amount\"`\n    IdAtGateway string `json:\"id_at_gateway\"`\n    Status enums.StatusEnum `json:\"status\"`\n    FraudFlag enums.FraudFlagEnum `json:\"fraud_flag\"`\n    InitiatorType enums.InitiatorTypeEnum `json:\"initiator_type\"`\n    ThreeDSecure bool `json:\"three_d_secure\"`\n    AuthorizationReason enums.AuthorizationReasonEnum `json:\"authorization_reason\"`\n    ErrorCode string `json:\"error_code\"`\n    ErrorText string `json:\"error_text\"`\n    VoidedAt *time.Time `json:\"voided_at\"`\n    ResourceVersion int64 `json:\"resource_version\"`\n    UpdatedAt *time.Time `json:\"updated_at\"`\n    FraudReason string `json:\"fraud_reason\"`\n    CustomPaymentMethodId string `json:\"custom_payment_method_id\"`\n    AmountUnused uint64 `json:\"amount_unused\"`\n    MaskedCardNumber string `json:\"masked_card_number\"`\n    ReferenceTransactionId string `json:\"reference_transaction_id\"`\n    RefundedTxnId string `json:\"refunded_txn_id\"`\n    ReferenceAuthorizationId string `json:\"reference_authorization_id\"`\n    AmountCapturable uint64 `json:\"amount_capturable\"`\n    ReversalTransactionId string `json:\"reversal_transaction_id\"`\n    Deleted bool `json:\"deleted\" validate:\"required\"`\n    MerchantReferenceId string `json:\"merchant_reference_id\"`\n    BusinessEntityId string `json:\"business_entity_id\"`\n    PaymentMethodDetails string `json:\"payment_method_details\"`\n    CustomPaymentMethodName string `json:\"custom_payment_method_name\"`\n    LinkedInvoices []InvoiceTransaction `json:\"linked_invoices\"`\n    InvoiceId string `json:\"invoice_id\" validate:\"required\"`\n    AppliedAmount uint64 `json:\"applied_amount\" validate:\"required\"`\n    AppliedAt *time.Time `json:\"applied_at\" validate:\"required\"`\n    InvoiceDate *time.Time `json:\"invoice_date\"`\n    InvoiceTotal uint64 `json:\"invoice_total\"`\n    InvoiceStatus enums.InvoiceStatusEnum `json:\"invoice_status\" validate:\"required\"`\n    LinkedCreditNotes []CreditNoteTransaction `json:\"linked_credit_notes\"`\n    CnId string `json:\"cn_id\" validate:\"required\"`\n    AppliedAmount uint64 `json:\"applied_amount\" validate:\"required\"`\n    AppliedAt *time.Time `json:\"applied_at\" validate:\"required\"`\n    CnReasonCode enums.CnReasonCodeEnum `json:\"cn_reason_code\"`\n    CnCreateReasonCode string `json:\"cn_create_reason_code\"`\n    CnDate *time.Time `json:\"cn_date\"`\n    CnTotal uint64 `json:\"cn_total\"`\n    CnStatus enums.CnStatusEnum `json:\"cn_status\" validate:\"required\"`\n    CnReferenceInvoiceId string `json:\"cn_reference_invoice_id\" validate:\"required\"`\n    LinkedRefunds []TxnRefundsAndReversal `json:\"linked_refunds\"`\n    TxnId string `json:\"txn_id\" validate:\"required\"`\n    TxnStatus enums.TxnStatusEnum `json:\"txn_status\" validate:\"required\"`\n    TxnDate *time.Time `json:\"txn_date\" validate:\"required\"`\n    TxnAmount uint64 `json:\"txn_amount\" validate:\"required\"`\n    LinkedPayments []LinkedPayment `json:\"linked_payments\"`\n    Id string `json:\"id\" validate:\"required\"`\n    Status enums.StatusEnum `json:\"status\"`\n    Amount uint64 `json:\"amount\"`\n    Date *time.Time `json:\"date\"`\n    RequestId string `json:\"request_id\"`\n    ErrorCategory string `json:\"error_category\"`\n    ErrorCode string `json:\"error_code\"`\n    ErrorMessage string `json:\"error_message\"`\n    DeclineCode string `json:\"decline_code\"`\n    DeclineMessage string `json:\"decline_message\"`\n    NetworkErrorCode string `json:\"network_error_code\"`\n    NetworkErrorMessage string `json:\"network_error_message\"`\n    ErrorField string `json:\"error_field\"`\n    RecommendationCode string `json:\"recommendation_code\"`\n    RecommendationMessage string `json:\"recommendation_message\"`\n    ProcessorErrorCode string `json:\"processor_error_code\"`\n    ProcessorErrorMessage string `json:\"processor_error_message\"`\n}"
'''
